### PR TITLE
Task 2 add student model

### PR DESCRIPTION
**Pull request by AbiOnMac**
## Description

Please include a summary of the changes and the issue that is being fixed. Please also include relevant motivation and context.

1. I have added the Student model with some attributes asked in (https://jobready.atlassian.net/wiki/spaces/RTO/pages/2878144513/Rails+Project+MySMS+Replicants+version) task list > task 2.
2. Factorbot rails and faker gems are added.
3. RSpec test for Student model is added.
4. Updated .gitignore file with "Application Database.session.sql"

Fixes # (issue)

## Type of change

Please select the relevant option.

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Other

## Checklist

Please review the following checklist and ensure all tasks are completed before submitting the pull request:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: spec/models/student_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    context "with duplicate email" do
+      let!(:existing_student) { FactoryBot.create(:student) }
+      let(:student_with_duplicate_email) { FactoryBot.build(:student, email: existing_student.email) }
+
+      it "is invalid" do
+        expect(student_with_duplicate_email).not_to be_valid
+        expect(student_with_duplicate_email.errors[:email]).to include("has already been taken")
+      end
+    end
+  end
```
Comment (josh-cooper) [inline spec/models/student_spec.rb]: This is missing specs for quite a few validations, such as last_name, birth_date, etc.

There's a common gem to simplify writing validation specs, we use it in Plus https://github.com/thoughtbot/shoulda-matchers

E.g.
```
describe 'validations' do
  it { should validate_presence_of(:first_name) }
  it { should_validate_uniqueness_of(:email) }
end
```

File: spec/factories/students.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    middle_name { name_parts[1..-2].join(" ") }
+    last_name { name_parts[-1] }
+    email { Faker::Internet.email }
+    birth_date { Faker::Date.birthday(min_age: 18, max_age: 65) }
+    gender { FAKE_GENDERS.sample }
+
+    # Adding traits in case of validations
+    trait :without_first_name do
+      first_name { nil }
+    end
```
Comment (josh-cooper) [inline spec/factories/students.rb]: This is the correct use of a trait, if a little superfluous. However it's not used anywhere so we can remove it.

**Review by josh-cooper — CHANGES_REQUESTED** [review]
Looking good but need to tighten up the specs

File: spec/models/student_spec.rb (hunk 1 above)
Comment (AbiOnMac) [reply spec/models/student_spec.rb]: @josh-cooper this is to demonstrate working of factory bot gem. Should I update it with shoulda-matchers? or just add it for missing validations

File: spec/models/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+RSpec.describe Student, type: :model do
+  context "validations" do
+    it { should validate_presence_of(:first_name) }
+    it { should validate_presence_of(:last_name) }
+    it { should validate_presence_of(:email) }
+    it { should validate_presence_of(:birth_date) }
+    it { should validate_presence_of(:gender) }
+    it { should validate_uniqueness_of(:email) }
+    it { should allow_value("valid@email.com").for(:email) }
+    it { should_not allow_value("invalid_email").for(:email) }
```
Comment (josh-cooper) [inline spec/models/student_spec.rb:12]: Nice 👍

**Review by josh-cooper — APPROVED** [review]
Lookin good

**Pull request merged**